qa-tests-backend/src/test/groovy/ProcessesListeningOnPortsTest.groovy

```diff
@@ -0,0 +1,252 @@
+    static final private String OTHER_NAMESPACE = "qa2"
+
+    static final private String SOCAT_DEBUG = "-d -d -v"
+
+    // Target deployments
+    @Shared
+    private List<Deployment> targetDeployments
+
+    def buildTargetDeployments() {
+        return [
+            new Deployment()
+                    .setName(TCPCONNECTIONTARGET1)
+                    .setImage("quay.io/rhacs-eng/qa:socat")
+                    .addPort(80)
+                    .addPort(8080)
+                    .addLabel("app", TCPCONNECTIONTARGET1)
+                    .setExposeAsService(true)
+                    .setCommand(["/bin/sh", "-c",])
+                    .setArgs(["(socat "+SOCAT_DEBUG+" TCP-LISTEN:80,fork STDOUT & " +
+                                      "socat "+SOCAT_DEBUG+" TCP-LISTEN:8080,fork STDOUT)" as String,]),
+            new Deployment()
+                    .setName(TCPCONNECTIONTARGET2)
+                    .setImage("quay.io/rhacs-eng/qa:socat")
+                    .addPort(8081, "TCP")
+                    .addLabel("app", TCPCONNECTIONTARGET2)
+                    .setExposeAsService(true)
+                    .setCommand(["/bin/sh", "-c",])
+                    .setArgs(["(socat "+SOCAT_DEBUG+" TCP-LISTEN:8081,fork STDOUT)" as String,]),
+        ]
+    }
+
+    def createDeployments() {
+        targetDeployments = buildTargetDeployments()
+        orchestrator.batchCreateDeployments(targetDeployments)
+        for (Deployment d : targetDeployments) {
+            assert Services.waitForDeployment(d)
+        }
+    }
+
+    def setupSpec() {
+        orchestrator.createNamespace(OTHER_NAMESPACE)
+        orchestrator.createImagePullSecret(
+                "quay",
+                Env.mustGet("REGISTRY_USERNAME"),
+                Env.mustGet("REGISTRY_PASSWORD"),
+                OTHER_NAMESPACE,
+                "https://quay.io"
+        )
+        orchestrator.createServiceAccount(
+                new K8sServiceAccount(
+                        name: "default",
+                        namespace: OTHER_NAMESPACE,
+                        imagePullSecrets: ["quay"]
+                )
+        )
+
+        createDeployments()
+    }
+
+    def destroyDeployments() {
+        for (Deployment deployment : targetDeployments) {
+            orchestrator.deleteDeployment(deployment)
+        }
+        for (Deployment deployment : targetDeployments) {
+            if (deployment.exposeAsService) {
+                orchestrator.waitForServiceDeletion(new Service(deployment.name, deployment.namespace))
+            }
+        }
+        orchestrator.deleteNamespace(OTHER_NAMESPACE)
+        orchestrator.waitForNamespaceDeletion(OTHER_NAMESPACE)
+    }
+
+    def cleanupSpec() {
+        destroyDeployments()
+    }
+
+    def rebuildForRetries() {
+        if (Helpers.getAttemptCount() > 1) {
+            log.info ">>>> Recreating test deployments prior to retest <<<<<"
+            destroyDeployments()
+            sleep(5000)
```

Comment:
do we need these 5 second sleeps here? Ideally, `destroyDeployments/createDeployments` will only ever return after ensuring the deployments are deleted / created.

Reply:
Removed